src/main/app/controller/courts/LocalAuthoritiesController.ts

```diff
@@ -0,0 +1,193 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {CourtType} from '../../../types/CourtType';
+import {
+  LocalAuthoritiesAreaOfLaw,
+  LocalAuthoritiesPageData,
+  LocalAuthority,
+  LocalAuthorityItem
+} from '../../../types/LocalAuthority';
+import {AreaOfLaw} from '../../../types/AreaOfLaw';
+
+import {SelectItem} from '../../../types/CourtPageData';
+import {CSRF} from '../../../modules/csrf';
+
+export enum familyAreaOfLaw {
+  adoption = 'Adoption',
+  children = 'Children',
+  civilPartnership = 'Civil partnership',
+  divorce = 'Divorce'
+}
+@autobind
+export class LocalAuthoritiesController {
+
+
+  getCourtTypesErrorMsg = 'A problem occurred when retrieving the court types.';
+  getCourtAreasOfLawErrorMsg = 'A problem occurred when retrieving the court areas of law. ';
+  getLocalAuthoritiesErrorMsg = 'A problem occurred when retrieving the list of local authorities. ';
+  getCourtLocalAuthoritiesErrorMsg = 'A problem occurred when retrieving the court local authorities. ';
+  updateErrorMsg = 'A problem occurred when saving the local authorities. ';
+  familyAreaOfLawErrorMsg = 'You need to enable relevant family court areas of law ';
+
+  public async getAreasOfLaw(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    error = '',
+    areasOfLaw: AreaOfLaw[] = null): Promise<void> {
+    const slug: string = req.params.slug as string;
+
+    if (!areasOfLaw ) {
+      await req.scope.cradle.api.getCourtAreasOfLaw(slug)
+        .then((value: AreaOfLaw[]) => areasOfLaw = value)
+        .catch(() => error += this.getCourtAreasOfLawErrorMsg);
+    }
+
+    if (areasOfLaw ){
+      areasOfLaw = this.checkFamilyAreasOfLaw(areasOfLaw);
+      if(!areasOfLaw.length){
+        error += this.familyAreaOfLawErrorMsg;
+      }
+    }
+
+    let courtTypes: CourtType[] = [];
+    await req.scope.cradle.api.getCourtCourtTypes(slug)
+      .then((value: CourtType[]) => courtTypes = value)
+      .catch(() => error += this.getCourtTypesErrorMsg);
+
+    const pageData: LocalAuthoritiesAreaOfLaw = {
+      errorMsg: error,
+      updated: updated,
+      isEnabled: !courtTypes.length ? false :courtTypes.some(c => c.name === 'Family Court'),
+      courtAreasOfLaw: this.getAreasOfLawForSelect(areasOfLaw, ''),
+    };
+
+    res.render('courts/tabs/localAuthoritiesContent', pageData);
+  }
+
+
+  public async getLocalAuthorities(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    error = '',
+    selectedAreaOfLaw = '',
```

Comment:
I don't think this variable is used in the method below. 

Closest thing to it is: 

```
    const pageData: LocalAuthoritiesPageData = {
      errorMsg: error,
      updated: updated,
      selectedAreaOfLaw: areaOfLaw, < this bit, which is areasOfLaw, not selectedAreaOfLaw
      courtAreasOfLaw: this.getAreasOfLawForSelect(areasOfLaw, areaOfLaw),
      localAuthoritiesItems: this.mapLocalAuthorityToLocalAuthorityItem(allLocalAuthorities, courtLocalAuthorities)
    };
```

Reply:
removed variable